app/models.py

```diff
@@ -929,37 +930,33 @@ def validates_data(self, key, data):
     def applications_closed_at(self):
         if self.published_at is None:
             return None
+        brief_publishing_date_and_length = self._build_date_and_length_data()
 
-        # Set time to 23:59:59 same day and add full number of days before application closes
-        published_day = self.published_at.replace(hour=23, minute=59, second=59, microsecond=0)
-        closing_time = published_day + timedelta(days=self.APPLICATIONS_OPEN_DAYS)
-
-        return closing_time
+        return get_publishing_dates(brief_publishing_date_and_length)['closing_date']
 
     @applications_closed_at.expression
     def applications_closed_at(cls):
-        return func.date_trunc('day', cls.published_at) + sql_cast(
-            '%d days 23:59:59' % cls.APPLICATIONS_OPEN_DAYS, INTERVAL
-        )
+        return sql_case([
+            (cls.data['requirementsLength'].astext == '1 week', func.date_trunc('day', cls.published_at) + sql_cast(
```

Comment:
There's a chance that `INTERVAL` understands weeks directly, might be worth checking if you haven't yet. Would make it easier to add additional lengths in the future.